wireapp/wire-android
# feat: Delete notification when message deleted (WPB-1669)

# What's new in this PR?

### Issues

When the sender edits/deletes a message, the notification on receiver side is not updated and displays the old message.

### Causes (Optional)

informing about message changes in notification was not implemented.

### Solutions

1. Update `kalium`
2. Update `MessageNotificationManager` to apply changes into an existed notifications, by getting displayed notifications, finding the specific message in it (by messageId), and applying changes. So if the message is not in the notification, or notification is not displayed - do nothing. 


Needs release with:

- https://github.com/wireapp/kalium/pull/1997

### Testing

#### How to Test

- as userA send a message to userB
- receive a notification on userB
- as a userA edit a message

expected result: notification for userB is updated


- as userA send a message to userB
- receive a notification on userB
- as a userA delete a message "for everyone"

expected result: notification for userB is removed (Including the summary notification)


- as userA send a few message to userB
- receive a notifications on userB
- as a userA delete 1 message "for everyone"

expected result: notification for userB is updated - deleted message is not displayed.
